build.gradle

```diff
@@ -19,35 +19,110 @@ plugins {
+    suppressionFile = 'dependency-check-suppressions.xml'
+    analyzers {
+        // Disable scanning of .NET related binaries
+        assemblyEnabled = false
+    }
+}
+
+compileJava {
+    options.compilerArgs << '-parameters' << '-Xlint:deprecation'
+}
+
+compileTestJava {
+    options.compilerArgs << '-Xlint:deprecation'
 }
 
 repositories {
-    maven { url "http://dl.bintray.com/hmcts/hmcts-maven/" }
     mavenLocal()
+    mavenCentral()
+    maven {
+        url "https://dl.bintray.com/hmcts/hmcts-maven"
+    }
+    maven {
+        url 'https://repo.spring.io/libs-milestone'
+    }
     jcenter()
 }
 
 dependencyManagement {
-	imports {
-		mavenBom org.springframework.boot.gradle.plugin.SpringBootPlugin.BOM_COORDINATES
-	}
+    imports {
+        mavenBom org.springframework.boot.gradle.plugin.SpringBootPlugin.BOM_COORDINATES
+    }
+
+    dependencies {
+        // checkstyle needs a certain version and so does spring fox swagger, so can't use latest
+        // fun times...
+        dependencySet(group: 'com.google.guava', version: '20.0') {
+            entry 'guava'
+        }
+        // solves CVE-2019-12086
+        // remove once spring manager incorporates this changes
+        dependencySet(
+                group: 'com.fasterxml.jackson.core',
+                version: '2.9.9'
+                //version: dependencyManagement.importedProperties['jackson.version'] solves CVE-2019-12086
+        ) {
+            entry 'jackson-core'
+            entry 'jackson-databind'
+        }
+        dependencySet(
+                group: 'com.fasterxml.jackson.datatype',
+                version: '2.9.9'
+                //version: dependencyManagement.importedProperties['jackson.version'] solves CVE-2019-12086
+        ) {
+            entry 'jackson-datatype-jdk8'
+            entry 'jackson-datatype-jsr310'
+        }
+        dependencySet(
+                group: 'com.fasterxml.jackson.module',
+                version: '2.9.9'
+                //version: dependencyManagement.importedProperties['jackson.version'] solves CVE-2019-12086
+        ) {
+            entry 'jackson-module-parameter-names'
+        }
+    }
+}
+
+def fileVersion = new org.yaml.snakeyaml.Yaml().load(new File("VERSION.yaml").newInputStream())
+
+group = 'uk.gov.hmcts.reform.cmc'
+version = fileVersion.TAG
+
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
+tasks.withType(JavaCompile) {
+    options.encoding = 'UTF-8'
 }
 
 dependencies {
-    annotationProcessor group: 'org.projectlombok', name: 'lombok', version: '1.18.6'
+    annotationProcessor group: 'org.projectlombok', name: 'lombok', version: '1.18.8'
 
+    compile group: 'org.springframework.boot', name: 'spring-boot-starter-json'
+    compile group: 'org.apache.commons', name: 'commons-lang3', version: '3.9'
     compile group: 'com.fasterxml.jackson.core', name: 'jackson-annotations'
+    compile group: 'org.springframework.boot', name: 'spring-boot-starter'
```

Comment:
Do we really need spring boot jars here?

Reply:
nope - ill remove the CCDAdapterConfig class - not required too